openwisp_controller/config/base/config.py

```diff
@@ -812,5 +812,63 @@ def manage_backend_changed(cls, instance_id, old_backend, backend, **kwargs):
             old_templates = device_group.templates.filter(backend=old_backend)
         config.manage_group_templates(templates, old_templates, not created)
 
+    def test_vpn_template_switch(self):
+        # Create a VPN server
+        vpn = self._create_vpn()
+
+        # Create two VPN templates using the same VPN server
+        template1 = self._create_template(
+            name='vpn-template-1',
+            type='vpn',
+            vpn=vpn,
+            auto_cert=True,
+            default=True,  # Auto-apply template1 to the device
+        )
+        template2 = self._create_template(
+            name='vpn-template-2',
+            type='vpn',
+            vpn=vpn,
+            auto_cert=True,
+        )
+
+        # Create a device and assign template1
+        device = self._create_device()
+        config = device.config
+        config.templates.add(template1)
+        config.save()
```

Comment:
We have used `default=True` while creating the template, thus we don't need to add the template1 to the Config object.